lib/asciidoctor/converter/semantic-html5.rb

```diff
@@ -29,6 +34,25 @@ def convert_paragraph node
     end
   end
 
+  def generate_header node
+    if node.header? && !node.noheader
```

Comment:
All combinations are covered:

| `header?` | `noheader` | `notitle` | Expected Result |
|-|-|-|-|
| `true` | `true` |  `true` | No `<header>` |
| `true` | `true` |  `false` |  No `<header>` |
| `true` | `false` |  `false` |  Complete `<header>` |
| `true` | `false` |  `true` |  `<header>` without title |
| `false` | `true` |  `true` |  No `<header>` |
| `false` | `true` |  `false` |  No `<header>` |
| `false` | `false` |  `true` |  No `<header>` |
| `false` | `false` |  `false` |  No `<header>` |

Reply:
I'll be honest, this whole notitle thing has always been really weird and ill-defined. And it clashes with noheader. The idea of it, originally, was to disable the display of the h1 title and only show the body of the document (almost always for embedded output). But that was before we had a well-defined header. And in PDF, it's meaning is to disable the title page, which is really the purpose the header is serving in HTML. I'm sort of leaning towards treating it like noheader, which is to just disable the `<header>` (or ignoring it).

Reply:
In HTML, it can make sense to hide the document title but still show document metadata (i.e., revision and author(s)). In PDF, not so much...

> And in PDF, it's meaning is to disable the title page, which is really the purpose the header is serving in HTML.

We also have the attribute `title-page` which serves the same purpose no?

Anyway, `notitle` is probably too fine grained and as you've mentioned it clashes with `noheader`. Also, if you want to hide the document title in HTML, it requires 1 line of CSS:

```css
header > h1 { display: none; }
```

> I'm sort of leaning towards treating it like noheader, which is to just disable the \<header> (or ignoring it).

When you say disable/ignore the `<header>`, you mean do no generate/produce the `<header>` element?